src/diffpy/utils/diffraction_objects.py

```diff
@@ -36,26 +36,67 @@ def _setter_wmsg(attribute):
 
 class DiffractionObject:
     def __init__(
-        self, name=None, wavelength=None, scat_quantity=None, metadata=None, xarray=None, yarray=None, xtype=None
+        self,
+        xarray,
+        yarray,
+        xtype,
+        wavelength,
+        scat_quantity="",
+        name="",
+        metadata={},
     ):
-        if name is None:
-            name = ""
-        self.name = name
-        if metadata is None:
-            metadata = {}
-        self.metadata = metadata
-        if xtype is None:
-            xtype = ""
+
+        self._id = uuid.uuid4()
+        self.input_data(xarray, yarray, xtype, wavelength, scat_quantity, name, metadata)
+
+    def input_data(self, xarray, yarray, xtype, wavelength, scat_quantity="", name="", metadata={}):
+        """
+        Insert a new scattering quantity into the scattering object.
+
+        Parameters
+        ----------
+        xarray : array-like
+            The independent variable array (e.g., "q", "tth", or "d").
+        yarray : array-like
+            The dependent variable array corresponding to intensity values.
+        xtype : str
+            The type of the independent variable in `xarray`. Must be one of {*XQUANTITIES},
+            such as "q", "tth", or "d".
+        wavelength : float
+            The wavelength of the incoming beam, specified in angstroms (Å).
+        scat_quantity : str, optional
+            The type of scattering experiment (e.g., "x-ray", "neutron"). Default is an empty string "".
+        name : str, optional
+            The name or label for the scattering data. Default is an empty string "".
+        metadata : dict, optional
+            The additional metadata associated with the diffraction object. Default is {}.
+
+        Returns
+        -------
+        None
+            This method updates the object in place and does not return a value.
+        """
+
+        # Check xtype is valid. An empty string is the default value.
+        if xtype not in XQUANTITIES:
+            raise ValueError(_xtype_wmsg(xtype))
+
+        # Check xarray and yarray have the same length
+        if len(xarray) != len(yarray):
+            raise ValueError(
+                "'xarray' and 'yarray' must have the same length. "
+                "Please re-initialize 'DiffractionObject' or re-run the method 'input_data' "
+                "with 'xarray' and 'yarray' of identical length."
+            )
+
         self.scat_quantity = scat_quantity
         self.wavelength = wavelength
+        self.metadata = metadata
+        self.name = name
 
-        if xarray is None:
-            xarray = np.empty(0)
-        if yarray is None:
-            yarray = np.empty(0)
-
-        self._id = uuid.uuid4()
-        self.input_data(xarray, yarray, xtype)
+        self._input_xtype = xtype
+        self._set_xarrays(xarray, xtype)
+        self._all_arrays[:, 0] = yarray
```

Comment:
It would be cleaner of this was moved into _set_arrays unless there was a reason not to do that because it is reused somewhere or something

Reply:
done